treeherder/webapp/api/performance_data.py

```diff
@@ -1015,15 +1014,20 @@ def list(self, request):
                     "platform": platform,
                     "base_app": base_sig.get("application", ""),
                     "new_app": new_sig.get("application", ""),
-                    "suite": base_sig.get("suite", ""),  # same suite for base_result and new_result
-                    "test": base_sig.get("test", ""),  # same test for base_result and new_result
+                    "suite": base_sig.get("suite", "")
+                    if base_sig
+                    else new_sig.get("suite", ""),  # same suite for base_result and new_result
+                    "test": base_sig.get("test", "")
+                    if base_sig
+                    else new_sig.get("test", ""),  # same test for base_result and new_result
                     "is_complete": is_complete,
                     "framework_id": framework,
-                    "is_empty": is_empty,
-                    "option_name": option_collection_map.get(
-                        base_sig.get("option_collection_id", ""), ""
-                    ),
-                    "extra_options": base_sig.get("extra_options", ""),
+                    "option_name": self._get_option_name(base_sig, option_collection_map)
+                    if base_sig
+                    else self._get_option_name(new_sig, option_collection_map),
+                    "extra_options": base_sig.get("extra_options", "")
+                    if base_sig
+                    else new_sig.get("extra_options", ""),
```

Comment:
What do you think about moving all of the `if base_sig` checks into an if statement before the `row_result` gets defined like on line 958? For example something like this (or maybe defining defaults as being the new_sig options, and setting them with base_sig options if `if base_sig` is true):

```
if base_sig:
    lower_is_better = ...
    
else:
    lower_is_better = ...
```